### PR TITLE
Use built-in CMake project version functionality

### PR DESCRIPTION
**Pull request by ChrisThrasher**
Thanks a lot for making a contribution to SFML! 🙂

Before you create the pull request, we ask you to check the follow boxes. (For small changes not everything needs to ticked, but the more the better!)

* [ ] Has this change been discussed on [the forum](https://en.sfml-dev.org/forums/index.php#c3) or in an issue before?
* [X] Does the code follow the SFML [Code Style Guide](https://www.sfml-dev.org/style.php)?
* [ ] Have you provided some example/test code for your changes?
* [ ] If you have additional steps which need to be performed list them as tasks!

----

## Description

CMake already has a notion of project version and various variables that encode it so let's use that. It's more idiomatic and requires less code.

## Tasks

* [ ] Tested on Linux
* [ ] Tested on Windows
* [x] Tested on macOS
* [ ] Tested on iOS
* [ ] Tested on Android

### CODE REVIEW

**Review by eXpl0it3r — APPROVED** [review]
Nice find! 🙂

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

**Comment by ChrisThrasher** [comment]
I think I fixed the failing MinGW builds.

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

File: CMakeLists.txt [hunk 1]
```diff
@@ -46,9 +46,6 @@ include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/Config.cmake)
 include(GNUInstallDirs)
 
 # setup version numbers
-set(VERSION_MAJOR 3)
-set(VERSION_MINOR 0)
-set(VERSION_PATCH 0)
 set(VERSION_IS_RELEASE OFF)
```
Comment (kimci86) [inline CMakeLists.txt:49]: This could be right after the `project` command to keep related things together.

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

File: CMakeLists.txt (hunk 1 above)
Comment (ChrisThrasher) [reply CMakeLists.txt]: Good idea. I moved that up.

**Pull request merged**